src/TestSuite/IntegrationTestCase.php

```diff
@@ -497,8 +506,37 @@ protected function _sendRequest($url, $method, $data = [])
             throw $e;
         } catch (Exception $e) {
             $this->_exception = $e;
-            $this->_handleError($e);
+            $this->_handleError($e, $psrRequest);
```

Comment:
I'm not sure we need to change this. This path is only followed when the exception middleware is not used. Because of that getting the current request out of `Router` should be good enough, as it more accurately reflects what would happen if the global default exception handler was invoked.

Reply:
I favor passing the $request along because you can mock it up in tests and for example test that some method was called at some point during the processing of the requesting. Would that sill work if using the `Router`?

Reply:
Outside of mocking this adds a place where the test harness and real behaviour could diverge again.